flocker/control/test/test_protocol.py

```diff
@@ -137,7 +141,41 @@ def build_control_amp_service(test):
                              TCP4ServerEndpoint(MemoryReactor(), 1234))
 
 
-class ControlAMPTests(SynchronousTestCase):
+class ControlTestCase(SynchronousTestCase):
+    """
+    Base TestCase for control tests that supplies a utility
+    method to patch the callRemote method of the AMP protocol instance,
+    discarding Eliot contexts whose context level may be unknown.
+    """
+
+    def patch_call_remote(self, capture_list, protocol=None):
+        """
+        Patch the callRemote method for this test case's protocol.
+
+        :param capture_list: A `list` to which results will be added.
+
+        :param protocol: Either `None` to default to self.protocol, or
+            a ``ControlAMP`` instance.
+        """
+        if protocol is None:
```

Comment:
:question: The tests are a lot more readable as a result of this, but I would suggest omitting this default and explicitly passing in `self.protocol`; too much succinctness makes things less understandable.